aspnetcore/security/anti-request-forgery.md

```diff
@@ -4,142 +4,232 @@ author: steve-smith
+1. A user signs into `www.good-banking-site.com` using forms authentication. The server authenticates the user and issues a response that includes an authentication cookie. The site is vulnerable to attack because it trusts any request that it receives with a valid authentication cookie.
+1. The user visits a malicious site, `www.bad-crook-site.com`.
 
-   The malicious site contains an HTML form similar to the following:
+   The malicious site, `www.bad-crook-site.com`, contains an HTML form similar to the following:
 
    ```html
-   <h1>You Are a Winner!</h1>
-   <form action="http://example.com/api/account" method="post">
+   <h1>Congratulations! You're a Winner!</h1>
+   <form action="http://good-banking-site.com/api/account" method="post">
        <input type="hidden" name="Transaction" value="withdraw">
        <input type="hidden" name="Amount" value="1000000">
-       <input type="submit" value="Click Me">
+       <input type="submit" value="Click to collect your prize!">
    </form>
    ```
 
-Notice that the form action posts to the vulnerable site, not to the malicious site. This is the “cross-site” part of CSRF.
+   Notice that the form's `action` posts to the vulnerable site, not to the malicious site. This is the "cross-site" part of CSRF.
 
-4. The user clicks the submit button. The browser automatically includes the authentication cookie for the requested domain (the vulnerable site in this case) with the request.
-5. The request runs on the server with the user's authentication context and can perform any action that an authenticated user is allowed to do.
+1. The user selects the submit button. The browser makes the request and automatically includes the authentication cookie for the requested domain, `www.good-banking-site.com`.
+1. The request runs on the `www.good-banking-site.com` server with the user's authentication context and can perform any action that an authenticated user is allowed to perform.
 
-This example requires the user to click the form button. The malicious page could:
+When the user selects the button to submit the form, the malicious site could:
 
 * Run a script that automatically submits the form.
 * Sends a form submission as an AJAX request. 
 * Use a hidden form with CSS. 
 
-Using SSL doesn't prevent a CSRF attack, the malicious site can send an `https://` request. 
+Using HTTPS doesn't prevent a CSRF attack. The malicious site can send an `https://www.good-banking-site.com/` request just as easily as it can send an insecure request.
+
+Some attacks target endpoints that respond to GET requests, in which case an image tag can be used to perform the action. This form of attack is common on forum sites that permit images but block JavaScript. Apps that change state on GET requests, where variables or resources are altered, are vulnerable to malicious attacks. **GET requests that changes state are insecure. A best practice is to never change state on a GET request.**
```

Comment:
that changes state --> that change state

Reply:
I split the blame with @Rick-Anderson on that one ... I copied his suggestion word-for-word.

Darn IT, Rick!!!! :smile: lol